staging/src/k8s.io/code-generator/cmd/client-gen/generators/fake/generator_fake_for_clientset.go

```diff
@@ -107,6 +118,37 @@ func (g *genClientset) GenerateType(c *generator.Context, t *types.Type, w io.Wr
 }
 
 // This part of code is version-independent, unchanging.
+
+var managedFieldsClientset = `
+func NewManagedFieldClientset(objects ...runtime.Object) *Clientset {
```

Comment:
I wish this had a name that emphasized more "this is the one new tests should use". Im not 100% comfortable with idea of it being opt-in, but get that there are a lot of deepequal checks on tracker output that would be broken by sudden appearance of managed fields, as well as tests dependent upon old broken behavior (which maybe should start breaking?)

Reply:
Exactly.  While working on this PR it's become clear to me how fragile tests are to _any_ change to the fake behavior.  And while I'd be OK with breaking in-tree usages, breaking out-of-tree usages is something I'd rather not do, at least not as part of this PR.  I'd be OK with renaming this to just `fake.NewClientset` and adding a deprecation notice to `NewSimpleClientset`. WDYT?